PCSX2/pcsx2
# [QT, I18N] Adding I18N comments for translators, minor typo fixes

### Description of Changes
This is a very first, very draftish, pass to add comments regarding specific, sporadic terms found on PCSX2. It needs a full review and explanations for any missing or TBW contents.

Any terms that are repeatedly mentioned throughout the emulator's texts should be added instead to Crowdin's internal glossary. I have left a first list on the #translators channel, in the Discord server.

QT's docs mention how to add comments in a way that they get exported to the .PO file that's used for Crowdin... (by adding a colon sign right after the // or /* signs) but it does not mention what method to do the same for the .ui files. For now, I just found a way to simply add comments, but I do not know if those comments will actually be printed to the .PO file. This requires tests done by programmers.

That brings me to the next point: I am not a programmer. I've Googled a bit and did some research on docs to get most of the terms, but I couldn't find out every term and I might have broken anything within the code. That is another reason this PR is a draft.

### Rationale behind Changes
 - Comments: to assist translators with their work, both in terms of knowledge and also so they know what genders/numbers should be used for each tech term/acronym.
 - Typo fixes: Bad Engrish is bad.

### Suggested Testing Steps
 - Read every comment and make sure it is accurate.
 - Make sure everything compiles properly.
 - Make sure every comment is exported to the translation file when applying Linguist.
 - Make sure the comments are not broken or breaking anything.
